File: src/main/java/novemberizing/rx/ds/tuple/Single.java
package novemberizing.rx.ds.tuple;

/**
 *
 * @author novemberizing, novemberizing@gmail.com
 * @since 2017. 1. 12.
 */
public class Single<A> extends Empty {
    public A first;

    public Single(){
        first = null;
    }

    public Single(A first){
        this.first = first;
    }
}

File: src/main/c++/novemberizing.old.3/ds/tuple/single.cc
#include "single.hh"

#ifdef    __NOVEMBERIZING_DS_TUPLE__SINGLE__EXAMPLE__

#include <iostream>

using namespace novemberizing;

int main(int argc, char ** argv) {

    ds::tuple::Single<int> single(1);

    std::cout << single.first << std::endl;

    return 0;
}

#endif // __NOVEMBERIZING_DS_TUPLE__SINGLE__EXAMPLE__
